src/Tracking/Events/Activated.php

```diff
@@ -0,0 +1,65 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tracking\Events;
+
+/**
+ * This class adds actions to track when the extension is activated.
+ * *
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tracking
+ */
+class Activated extends BaseEvent {
+
+	/**
+	 * The page where activation with a source can occur.
+	 */
+	public const ACTIVATION_PAGE = 'plugin-install.php';
+
+	/**
+	 * The query parameters used to determine activation source details.
+	 */
+	public const SOURCE_PARAMS = [
+		'traffic_source',
+		'traffic_type',
+	];
+
+	/**
+	 * Register the tracking class.
+	 */
+	public function register(): void {
+		add_action( 'woocommerce_gla_extension_activated', [ $this, 'maybe_track_activation_source' ] );
+	}
+
+	/**
+	 * Track when the extension is activated from a source.
+	 *
+	 * @param array $server_vars The server variables to check for source information.
+	 */
+	public function maybe_track_activation_source( array $server_vars = [] ): void {
+		// Default to $_SERVER
+		if ( empty( $server_vars ) ) {
+			$server_vars = $_SERVER;
+		}
+
+		// Skip WP-CLI activations
+		if ( empty( $server_vars['HTTP_REFERER'] ) ) {
+			return;
+		}
+
+		$url_components = wp_parse_url( $server_vars['HTTP_REFERER'] );
+		// Only continue for Add Plugins page activations
+		if ( false === strstr( $url_components['path'], self::ACTIVATION_PAGE ) || ! $url_components || empty( $url_components['query'] ) ) {
```

Comment:
The `wp_parse_url` seems to have many different return types. Should we check if it's an array and contains the `path` component?

Reply:
Seems like it _should_ only be `false` or an array since no `component` param is provided, so I've added a few extra checks in 40087c03ad5f6d709ca2452a9b8f03c88660450c